### PR TITLE
ROECCT-245- Add New field for legal entity beneficiaries to indicate became dates.

### PR DESCRIPTION
**Pull request by krenjith-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROECCT-245

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/model/trust.model.ts [hunk 1]
```diff
@@ -39,6 +39,9 @@ export interface Trust {
   HISTORICAL_BO?: TrustHistoricalBeneficialOwner[];
   CORPORATES?: TrustCorporate[];
   review_status?: TrustReviewStatus;
+  start_date_day?: string;
+  start_date_month?: string;
+  start_date_year?: string;
```
Comment (ConnorOS-ch) [inline src/model/trust.model.ts:44]: Does the Trust itself need a start date? I thought this PR was only to add start date to Trust legal entity beneficiaries

File: src/model/trust.page.model.ts [hunk 2]
```diff
@@ -15,6 +15,9 @@ type TrustDetailsForm = {
   stillInvolved?: string;
   beneficialOwnersIds: string[];
   hasAllInfo: string;
+  startDateDay?: string;
+  startDateMonth?: string;
+  startDateYear?: string;
```
Comment (ConnorOS-ch) [inline src/model/trust.page.model.ts:20]: Same comment above

File: views/includes/page-templates/trust-legal-entity-bo.html [hunk 3]
```diff
@@ -139,7 +139,20 @@ <h1 class="govuk-heading-xl govuk-!-margin-0">{{ pageParams.title }}</h1>
...
+      name: 'startDate',
+      value: {
+        'Day': formData.startDateDay,
+        'Month': formData.startDateMonth,
+        'Year': formData.startDateYear
+      },
+      label: "When did they become a beneficiary?",
+      error: errors.startDate if errors
+    } %}
+    {% include "includes/inputs/date-input.html" %}
```
Comment (ConnorOS-ch) [inline views/includes/page-templates/trust-legal-entity-bo.html:154]: maybe we can use `includes/inputs/date/start-date-input.html` instead? check out what they have done for start date for beneficial owners

**Review by ConnorOS-ch — CHANGES_REQUESTED** [review]
Looks good, just a few comments. I think we can remove start date for the trust object and leave it for Trust legal entitity

File: src/model/trust.page.model.ts (hunk 2 above)
Comment (krenjith-ch) [reply src/model/trust.page.model.ts]: done

**Review by ConnorOS-ch — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**